pkg/proxy/proxy.go

```diff
@@ -1,267 +0,0 @@
-			return false
-		}
-	}
-
-	if h.Config.Authentication.Header.Enabled {
-		// Seemingly well-known headers to tell the upstream about user's identity
-		// so that the upstream can achieve the original goal of delegating RBAC authn/authz to kube-rbac-proxy
-		headerCfg := h.Config.Authentication.Header
-		req.Header.Set(headerCfg.UserFieldName, u.User.GetName())
-		req.Header.Set(headerCfg.GroupsFieldName, strings.Join(u.User.GetGroups(), headerCfg.GroupSeparator))
-	}
-
-	return true
-}
-
-func newKubeRBACProxyAuthorizerAttributesGetter(authzConfig *authz.Config) *krpAuthorizerAttributesGetter {
-	return &krpAuthorizerAttributesGetter{authzConfig}
-}
-
-type krpAuthorizerAttributesGetter struct {
-	authzConfig *authz.Config
-}
-
-// GetRequestAttributes populates authorizer attributes for the requests to kube-rbac-proxy.
-func (n krpAuthorizerAttributesGetter) GetRequestAttributes(u user.Info, r *http.Request) []authorizer.Attributes {
-	apiVerb := ""
-	switch r.Method {
-	case "POST":
-		apiVerb = "create"
-	case "GET":
-		apiVerb = "get"
-	case "PUT":
-		apiVerb = "update"
-	case "PATCH":
-		apiVerb = "patch"
-	case "DELETE":
-		apiVerb = "delete"
-	}
-
-	var allAttrs []authorizer.Attributes
-
-	defer func() {
-		for attrs := range allAttrs {
-			klog.V(5).Infof("kube-rbac-proxy request attributes: attrs=%#+v", attrs)
-		}
-	}()
-
-	if n.authzConfig.ResourceAttributes == nil {
-		// Default attributes mirror the API attributes that would allow this access to kube-rbac-proxy
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       "",
-			APIGroup:        "",
-			APIVersion:      "",
-			Resource:        "",
-			Subresource:     "",
-			Name:            "",
-			ResourceRequest: false,
-			Path:            r.URL.Path,
-		})
-		return allAttrs
-	}
-
-	if n.authzConfig.Rewrites == nil {
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       n.authzConfig.ResourceAttributes.Namespace,
-			APIGroup:        n.authzConfig.ResourceAttributes.APIGroup,
-			APIVersion:      n.authzConfig.ResourceAttributes.APIVersion,
-			Resource:        n.authzConfig.ResourceAttributes.Resource,
-			Subresource:     n.authzConfig.ResourceAttributes.Subresource,
-			Name:            n.authzConfig.ResourceAttributes.Name,
-			ResourceRequest: true,
-		})
-		return allAttrs
-	}
-
-	params := []string{}
-	if n.authzConfig.Rewrites.ByQueryParameter != nil && n.authzConfig.Rewrites.ByQueryParameter.Name != "" {
-		if ps, ok := r.URL.Query()[n.authzConfig.Rewrites.ByQueryParameter.Name]; ok {
-			params = append(params, ps...)
-		}
-	}
-	if n.authzConfig.Rewrites.ByHTTPHeader != nil && n.authzConfig.Rewrites.ByHTTPHeader.Name != "" {
-		if p := r.Header.Get(n.authzConfig.Rewrites.ByHTTPHeader.Name); p != "" {
-			params = append(params, p)
-		}
-	}
-
-	if len(params) == 0 {
```

Comment:
This is a case where the configuration means: "forward this request without performing authorization"?  I need to check the config file format, but I'm surprised that the default is "don't protect".  Am I reading this incorrectly?

Reply:
Well, the logic is not trivial and begs to break on code changes, but it looks like so:

1. We start with [empty allAttr](https://github.com/brancz/kube-rbac-proxy/blob/master/pkg/proxy/proxy.go#L143=).
2. The next two if-blocks are returning.
3. The next two if-blocks fill the  `params`.
4. If `params` is empty, return `allAttrs`, which is by then still empty.
5. And [here](https://github.com/brancz/kube-rbac-proxy/blob/master/pkg/proxy/proxy.go#L83-L84=) we fail, if the returned `allAttrs` is empty.

Reply:
TODO: pre-acceptance this should proactively error from this method so that we aren't succeeding by coordination in two unrelated parts.